persistence-modules/hibernate-mapping/src/test/java/com/baeldung/hibernate/wherejointable/HibernateWhereJoinTableIntegrationTest.java

```diff
@@ -0,0 +1,118 @@
+    private Session session;
+
+    /**
+     * Test data
+     */
+    private User user1;
+    private User user2;
+    private User user3;
+    private Group group1;
+    private Group group2;
+    
+    public void saveRelation(User user, Group group, UserGroupRole role) {
+
+        UserGroupRelation relation = new UserGroupRelation(user.getId(), group.getId(), role);
+        
+        session.save(relation);
+        session.flush();
+        session.refresh(user);
+        session.refresh(group);
+    }
+    
+    @BeforeClass
+    public static void beforeTests() {
+        Configuration configuration = new Configuration()
+            .addAnnotatedClass(User.class)
+            .addAnnotatedClass(Group.class)
+            .addAnnotatedClass(UserGroupRelation.class)
+            .setProperty("hibernate.dialect", H2Dialect.class.getName())
+            .setProperty("hibernate.connection.driver_class", org.h2.Driver.class.getName())
+            .setProperty("hibernate.connection.url", "jdbc:h2:mem:test")
+            .setProperty("hibernate.connection.username", "sa")
+            .setProperty("hibernate.connection.password", "")
+            .setProperty("hibernate.hbm2ddl.auto", "update");
+        ServiceRegistry serviceRegistry = new StandardServiceRegistryBuilder()
+            .applySettings(configuration.getProperties())
+            .build();
+        sessionFactory = configuration.buildSessionFactory(serviceRegistry);
+    }
+
+    @Before
+    public void setUp() {
+        session = sessionFactory.openSession();
+        session.beginTransaction();
+        
+        user1 = new User("user1");
+        user2 = new User("user2");
+        user3 = new User("user3");
+
+        group1 = new Group("group1");
+        group2 = new Group("group2");
+
+        session.save(group1);
+        session.save(group2);
+
+        session.save(user1);
+        session.save(user2);
+        session.save(user3);
+
+        saveRelation(user1, group1, UserGroupRole.MODERATOR);
+        saveRelation(user2, group1, UserGroupRole.MODERATOR);
+        saveRelation(user3, group1, UserGroupRole.MEMBER);
+
+        saveRelation(user1, group2, UserGroupRole.MEMBER);
+        saveRelation(user2, group2, UserGroupRole.MODERATOR);
+    }
+
+    @Test
+    public void givenUser1_getGroups_returnsAllGroups() {
+        List<Group> groups = user1.getGroups();
+        assertEquals(2, groups.size());
+
+        assertTrue(groups.contains(group1));
+        assertTrue(groups.contains(group2));
+    }
+
+    @Test
+    public void givenUser1_getModeratorGroups_returnsOnlyModeratorGroups() {
+        List<Group> groups = user1.getModeratorGroups();
+        assertEquals(1, groups.size());
+
+        assertTrue(groups.contains(group1));
+    }
+    
+    @After
+    public void tearDown() {
+        session.getTransaction().commit();
+        session.close();
+    }
+
+    @AfterClass
+    public static void afterTests() {
+        sessionFactory.close();
+    }
```

Comment:
I think this test class has its methods in an unconventional order. It's most common for the order to be something like:

- befores
- afters
- tests
- private helper methods

This class seems to be

- a helper method
- befores
- tests
- afters

Reply:
I changed them to the conventional order as you said.